src/ray/core_worker/reference_count.cc

```diff
@@ -1003,6 +1006,38 @@ absl::optional<LocalityData> ReferenceCounter::GetLocalityData(
   return locality_data;
 }
 
+void ReferenceCounter::PushToLocationSubscribers(ReferenceTable::iterator it) {
+  auto callbacks = it->second.location_subscription_callbacks;
+  it->second.location_subscription_callbacks.clear();
+  it->second.location_version++;
+  for (const auto callback : callbacks) {
+    callback(it->second.locations, it->second.object_size, it->second.location_version);
+  }
+}
+
+Status ReferenceCounter::GetObjectLocationsAsync(
+    const ObjectID &object_id, int64_t last_location_version,
+    const LocationSubscriptionCallback &callback) {
+  absl::MutexLock lock(&mutex_);
+  auto it = object_id_refs_.find(object_id);
+  if (it == object_id_refs_.end()) {
+    RAY_LOG(WARNING) << "Tried to register a location subscriber for an object "
```

Comment:
Can you explain when this can happen or add RAY_CHECK instead?

Reply:
This happens if the object has already been freed, this should actually probably be at the `INFO` level I think. Definitely shouldn't be a `RAY_CHECK`.